### PR TITLE
Update for Kaggle Dockerfile

### PR DESCRIPTION
**Pull request by pdmack**
Smoke-tested version pushed to gcr.io/kubeflow-dev/kubeflow-kaggle-notebook:latest

/assign @jlewi

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/2105)

### CODE REVIEW

File: components/contrib/kaggle-notebook-image/Dockerfile [hunk 1]
```diff
@@ -41,9 +42,8 @@ RUN export CLOUD_SDK_REPO="cloud-sdk-$(lsb_release -c -s)" && \
     curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add - && \
     apt-get update && \
     apt-get install -y google-cloud-sdk kubectl && \
-    # pin to 0.8.1 due to conflicts with tornado>=5.x and pyzmq>=17.x
     # we just really need jupyterhub-singleuser for our KF scripts
-    pip install jupyterhub==0.8.1 \
+    pip install jupyterhub \
```
Comment (jlewi) [inline components/contrib/kaggle-notebook-image/Dockerfile]: Why not pin to a particular version of jupyterhub so it doesn't break unexpectedly?

**Comment by pdmack** [comment]
`gcr.io/kubeflow-dev/kubeflow-kaggle-notebook:latest`

File: components/contrib/kaggle-notebook-image/Dockerfile [hunk 2]
```diff
@@ -41,9 +42,8 @@ RUN export CLOUD_SDK_REPO="cloud-sdk-$(lsb_release -c -s)" && \
     curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add - && \
     apt-get update && \
     apt-get install -y google-cloud-sdk kubectl && \
-    # pin to 0.8.1 due to conflicts with tornado>=5.x and pyzmq>=17.x
     # we just really need jupyterhub-singleuser for our KF scripts
-    pip install jupyterhub==0.8.1 \
+    pip install jupyterhub==0.9.6 \
```
Comment (jlewi) [inline components/contrib/kaggle-notebook-image/Dockerfile]: Should we just remove jupyterhub now that we are no longer using jupyterhub in Kubeflow?

**Comment by pdmack** [comment]
This from the guy who wants to put emacs in the images?

**Comment by pdmack** [comment]
There is a trailing adoption of Kubeflow and there will be users who have potentially invested at the earlier 0.3 and 0.4 releases. IMHO, we should keep jupyterhub in the images for some period of time since afaik it provides a degree of backward compatibility with no penalty that I'm aware of.

**Comment by jlewi** [comment]
Is installing JupyterHub sufficient though for maintaining compatibility with JupyterHub?

/lgtm
/approve

**Pull request merged**